Storage/Storage/Model/WooCommerce.xcdatamodeld/Model 27.xcdatamodel/contents

```diff
@@ -0,0 +1,554 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```

Comment:
I think we also need to update the deletion rules here. 

My understanding of Core Data (especially deletion rules) is very limited so I'm gonna state how I understand the current configuration. 

## My Understanding of the Current Config (Model 27)

### Product to Category

![image](https://user-images.githubusercontent.com/198826/78407591-6bbee800-75c2-11ea-8784-ec71cd05a0f7.png)

The way I understand this is that if the product is deleted, the `categories` should be deleted too. 

### Category to Product

![image](https://user-images.githubusercontent.com/198826/78407670-9b6df000-75c2-11ea-998d-d91c261abba6.png)

I understand this as "If a category is deleted, remove that category from the linked Products' categories". 

## What I Think it Should be

With this new product category synchronization, I think deleting the categories when a product is deleted is not ideal. It should probably be the other way around. However, it's probably also wrong to delete the products if a category is deleted. 

Maybe the rules should be:

- If a product is deleted, the category should be left alone. We don't want to delete the category anymore because we will use it in the picker. 
- If a category is deleted and there are no linked products, it **can** be deleted.
- If a category is deleted and **there are** linked products, the deletion should be **denied**. My concern is that if we use: 
  - `Nullify`: We might end up with a product that is _inaccurate_ (missing category). 
  - `Cascade`: We might delete a product that is currently being edited.

---

Did I understand this correctly? 😅 What do you think about these rules?

@jaclync and @pmusolino Would you mind sharing your thoughts too? 🙂 

@astralbodies Maybe you have some background knowledge related to this?

Reply:
I totally forgot about deletion rules 🤦‍♂
After a quick read on them I agree with your assumptions, I've updated them to:
```
Product -> Nullify -> categories
category -> Deny -> products
```
https://github.com/woocommerce/woocommerce-ios/pull/2079/commits/dafea866dc3554c1c49fc76e2d3dfafa1592598b

But I'm also curios to hear about other members thoughts 😇

Reply:
sorry for the delay! from my understanding, I think the deletion rules are like:

- `Product --(to-many)--> Set<ProductCategory>`: Nullify (when a product category is deleted, it just removes itself from the product's to-many relationship, since a product could have multiple categories)
- `ProductCategory --(to-many)--> Set<Product>`: Nullify (when a product is deleted, it just removes itself from the product category's to-many relationship, since many products can share the same category)

lemme know if I missed anything though!

Reply:
Hey @jaclync, thanks for your response! 🙇 

> Product --(to-many)--> Set<ProductCategory>: Nullify (when a product category is deleted, it just removes itself from the product's to-many relationship, since a product could have multiple categories)

I've just checked how the web-app works and it does follow that rule. If a product category is deleted, the deletion is allowed and it's removed from the products's category list.

@shiki Are you ok for us to follow that rule?

Reply:
@Ecarrion Yeah, that sounds fine to me. The Deny could also be tricky to code around so I'm okay with Nullify for now. 

Let me know if you've made the changes and also how we should migrate to this new structure.

Reply:
Sorry for the delay, but I see the ping only 2 minutes ago.
I think that what @jaclync said is correct 🙇